### PR TITLE
create health checks form component

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes**
https://issues.redhat.com/browse/ODC-3333

**Solution Description**
- Created a Form Component for addition and removal of readiness, liveness and startup probes. 
- Created a Formik field component for InputGroups
- Created a component to take an array of strings as input

**GIF**
HCC

### CODE REVIEW

File: frontend/packages/console-shared/src/components/formik-fields/InputField.tsx [hunk 1]
```diff
@@ -11,12 +17,29 @@ const InputField: React.FC<InputFieldProps> = ({
   onChange,
   validated,
   helpTextInvalid,
+  inputGroupText,
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/InputField.tsx:20]: Adding just a string prop for input group text doesn't really cover all of the scenarios an `InputGroup` might be used. We have so many patterns that are supported by an `InputGroup` https://www.patternfly.org/v4/documentation/react/components/inputgroup. But all these patterns are not supported by this `FormikField`. 

It would be better if you create another field called `InputGroupField` that handles all these scenarios. Also look at the implementation of `CheckBoxField` and `SwitchField` that use the same logic from `ToggleableFieldBase`. `InputField` and `InputGroupField` can follow similar logic of code sharing.

File: frontend/packages/console-shared/src/components/formik-fields/field-types.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ export interface InputFieldProps extends FieldProps {
   placeholder?: string;
   onChange?: (event) => void;
   onBlur?: (event) => void;
+  inputGroupText?: string;
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/field-types.ts:24]: This should support `ReactNode` and not just `string`.

File: frontend/packages/console-shared/src/components/form-utils/FormFooter.tsx [hunk 3]
```diff
@@ -34,11 +38,17 @@ const FormFooter: React.FC<FormFooterProps> = ({
         {infoMessage}
       </Alert>
     )}
-    <ActionGroup className="pf-c-form pf-c-form__group--no-top-margin">
+    <ActionGroup
+      className={cx(
+        'pf-c-form',
+        { 'pf-c-form__group--no-top-margin': noTopMargin },
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/form-utils/FormFooter.tsx:44]: It would be better if you just pass a prop from the parent called `actionBarClassNames` and merge them here.

File: frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  ReadinessProbe = 'readinessProbe',
+  LivenessProbe = 'livenessProbe',
+  StartupProbe = 'startupProbe',
+}
+
+export enum ProbeType {
+  'HTTP GET' = 'HTTP GET',
+  'Container Command' = 'Container Command',
+  'TCP Socket' = 'TCP Socket',
+}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts:11]: Why enum keys are string and have space?
```suggestion
export enum ProbeType {
  httpGet = 'HTTP GET',
  containerCommand = 'Container Command',
  tcpSocket = 'TCP Socket',
}
```

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          helpText="How long to wait for the probe to finish, if the time is exceeded, the probe is considered failed."
+          inputGroupText="seconds"
+          style={{ maxWidth: '100%' }}
+        />
+        <InputField
+          type={TextInputTypes.number}
+          name="successThreshold"
+          label="Success Threshold"
+          style={{ maxWidth: '100%' }}
+        />
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:88]: All other input fields have help text. Not for this one?

File: frontend/packages/console-shared/src/components/form-utils/FormFooter.tsx [hunk 6]
```diff
@@ -20,6 +20,9 @@ const FormFooter: React.FC<FormFooterProps> = ({
   disableSubmit,
   showAlert,
   sticky,
+  actionBarClassName = 'pf-c-form__group--no-top-margin',
+  submitButtonVariant = ButtonVariant.primary,
+  cancelButtonVariant = ButtonVariant.secondary,
```
[the remark this answers is not in the document]
Comment (rohitkrai03) [reply frontend/packages/console-shared/src/components/form-utils/FormFooter.tsx:25]: I agree with Sahil here. Although this is not just a use case with Health Check forms, we do have more forms with this kind of form footer in them. Which makes me think why not have two different variants of the `FormFooter` itself? The logic in the current `FormFooter` component can be smartly shared with two different variants. The main API should remain unchanged and simply a new prop called `variant` can define what variant is going to be rendered. That way we don't have to deal with controlling so many props from the parent form to get the desired form footer.

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  errors,
+  containerPorts,
+}) => {
+  const { values, setFieldValue } = useFormikContext<FormikValues>();
+  const onProbeTypeChange = (value) => {
+    setFieldValue('probeType', value);
+  };
+  return (
+    <div className="odc-heath-check__probe-form">
+      <FormSection>
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:49]: No, this FormSection will not have a title

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx (hunk 5 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:94]: I have worked around the High-Level Design, I would need UX's input on this one.

File: frontend/packages/console-shared/src/components/form-utils/form-utils-types.ts [hunk 8]
```diff
@@ -13,4 +13,10 @@ export interface FormFooterProps {
   showAlert?: boolean;
   infoTitle?: string;
   infoMessage?: string;
+  formFooterVariant?: FormFooterVariant;
+}
+
+export enum FormFooterVariant {
+  Default = 'defaultVariant',
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/form-utils/form-utils-types.ts:20]:
```suggestion
  Default = 'Default',
```

File: frontend/packages/console-shared/src/components/formik-fields/InputField.tsx [hunk 9]
```diff
@@ -1,46 +1,15 @@
...
-        aria-describedby={`${fieldId}-helper`}
-        value={field.value || ''}
-        onChange={(value, event) => {
-          field.onChange(event);
-          onChange && onChange(event);
-        }}
-      />
-    </FormGroup>
-  );
+export type InputFieldProps = BaseInputFieldProps & {
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/InputField.tsx:7]: Why not create this as an interface and extend `BaseInputFieldProps` in `field-types` file itself like all the other types defined there?

File: frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-unused-vars, no-undef */
+import * as React from 'react';
+import { InputGroup, TextInput, TextArea, TextInputTypes } from '@patternfly/react-core';
+import { GroupInputProps, BaseInputFieldProps, TextType } from './field-types';
+import BaseInputField from './BaseInputField';
+
+export type InputGroupFieldProps = GroupInputProps &
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx:7]: Same as above. Why not just extend the type in field-types file. You already have a declaration there for `InputGroupField` in that file.

File: frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  BaseInputFieldProps & {
+    type?: TextInputTypes;
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx:9]: When type is needed in both `InputField` and `InputGroupField` then why did you remove it from `BaseInputField` and adding it here explicitly?

File: frontend/packages/console-shared/src/components/formik-fields/field-types.ts [hunk 12]
```diff
@@ -16,25 +16,35 @@ export interface FieldProps {
...
 
-export interface InputFieldProps extends FieldProps {
-  type: TextInputTypes;
+export interface BaseInputFieldProps extends FieldProps {
   placeholder?: string;
   onChange?: (event) => void;
   onBlur?: (event) => void;
 }
 
+export interface GroupInputProps {
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/field-types.ts:25]:
```suggestion
export interface GroupInputProps extends BaseInputFieldProps {
```

File: frontend/packages/console-shared/src/components/formik-fields/field-types.ts [hunk 13]
```diff
@@ -16,25 +16,35 @@ export interface FieldProps {
...
+  beforeInput?: React.ReactNode;
+  afterInput?: React.ReactNode;
+  textType?: TextType;
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/field-types.ts:28]: Better name it `groupInputType` since we already have a `TextInputType`. Makes it less confusing.

File: frontend/packages/console-shared/src/components/formik-fields/field-types.ts [hunk 14]
```diff
@@ -16,25 +16,35 @@ export interface FieldProps {
...
+  TextArea = 'textArea',
+}
+
 export interface CheckboxFieldProps extends FieldProps {
   formLabel?: string;
   value?: string;
 }
 
-export interface SearchInputFieldProps extends InputFieldProps {
+export interface SearchInputFieldProps extends BaseInputFieldProps {
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/field-types.ts:47]: This will also need a `type` prop but you've removed it from the base type interface.

File: frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx (hunk 11 above)
Comment (debsmita1) [reply frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx:9]: @rohitkrai03  The reason behind doing this is, for InputField I want to keep the `type` mandatory & for `InputGroupField` want it optional because users can have both TextArea and Text and as we know for TextArea user don't need type. Let me know if there's a better way to do this

Comment (rohitkrai03) [reply frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx:9]: You can make type optional with having text as a default for InputField. Else just make use of the extend mechanism in the field-types file like all the other props defined there.

Comment (debsmita1) [reply frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx]: made the `type` optional with text as a default for InputField

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+  };
+
+  const handleAddProbe = () => {
+    setAddProbe(true);
+  };
+
+  const renderProbe = () => {
+    if (addProbe) {
+      return (
+        <Formik
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:62]: @debsmita1 I dont' think we need this ```Formik``` here for probe form. cc: @rohitkrai03

File: frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+              placeholder={placeholder}
+              isReadOnly={isReadOnly}
+            />
+          </FlexItem>
+          <FlexItem>
+            {!isReadOnly && (
+              <MinusCircleIcon
+                aria-hidden="true"
+                style={{ color: values.length === 1 ? disabledColor.value : null }}
+                onClick={(e) => {
```
Comment (vikram-raj) [inline frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx:53]: Unused variable ```e```.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+import { getHealthChecksProbeData } from './health-checks-utils';
+
+interface HealthChecksProps {
+  pageTitle?: string;
+}
+
+const renderProbes = () => {
+  return (
+    <>
+      <HealthCheckProbe
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:15]: Why do you have  three `Formik` instances for this form here? How would you integrate the forms inside the actual form or advanced options of git import if you have created separate formik instances of each section?

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx (hunk 15 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:62]: How would you integrate the forms inside the actual form or advanced options of git import if you have created separate formik instances of each section?

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+    setSelectedProbe('');
+    setProbeData(initialValues);
+  };
+
+  const handleReset = (values, actions) => {
+    actions.resetForm({ values: initialValues, status: {} });
+    setAddProbe(false);
+  };
+
+  const handleSubmit = (values, actions) => {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:37]: I can see that your `handleSubmit` doesn't actually do anything apart from just setting some local state. How do you plan to integrate this into an actual form that creates some resources based on the data entered by users? How does the parent form get access to the data inside this inner form?

File: frontend/packages/dev-console/src/components/health-checks/health-checks-utils.ts [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    tcpSocket: {
+      port: '8080',
+    },
+    command: [''],
+  };
+  return probeTypeData;
+};
+
+export const getHealthChecksProbeData = (probe: string): HealthCheckProbeDataType => {
+  const healthChecksProbeData = {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/health-checks-utils.ts:20]: Your health check probe data contains exact same object for each of the types and you're sending the same data in all the cases. Why do you even need this util function here?

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+
+  const handleAddProbe = () => {
+    setAddProbe(true);
+  };
+
+  const renderProbe = () => {
+    if (addProbe) {
+      return (
+        <Formik
+          initialValues={!_.isEmpty(probeData) ? probeData : initialValues}
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:63]: So your `initialValues` changes based on the user entered values? What does that do exactly?

File: frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+import { InputField, useFormikValidationFix } from '@console/shared';
+import MultiColumnFieldFooter from './multi-column-field/MultiColumnFieldFooter';
+import { getFieldId } from './field-utils';
+import { FieldProps } from './field-types';
+
+export interface TextColumnFieldProps extends FieldProps {
+  required?: boolean;
+  isValid?: boolean;
+  name: string;
+  values: string[];
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx:15]: Why do you need to take values as a prop here? A field component is supposed to connect to the formik context directly using just the name of the field and `useField` hook.

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx (hunk 18 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:37]: @rohit while integrating the form component in the Advanced section, I am doing a setFieldValue on probe form submission

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:37]: And how would that again work inside your actual health check form?

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:37]: If you're using `setFieldValue` inside you're handleSubmit for the inner formik, that means you have the same data present in initialValues of the outer formik as well? Then why do you actually need to create new formik instances for each of these sections? It should have been a simple form section component instead of a new `Formik` every time.

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx (hunk 15 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:62]: `Formik` inside of `Formik` is not a usual pattern that we want to follow unless its absolutely necessary. In cases where you want the form to be actually submitted and create some resources or something like we do with create secrets modal, then its okay. But not in this case.

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx (hunk 20 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:63]: I did this to preview the entered values when user clicks on `Edit Probe`

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx:63]: That is the issue, you're using wrong patterns to do these things. `initialValues` of the form should not change like that.

File: frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.scss [hunk 22]
```diff
@@ -0,0 +1,4 @@
+.odc-heath-check-probe__formTitle {
+  margin: 0px;
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.scss:2]:
```suggestion
  margin: 0;
```

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+      <HealthCheckProbe probe={HealthChecksProbeType.StartupProbe} />
+    </>
+  );
+};
+
+const HealthChecks: React.FC<HealthChecksProps> = ({ pageTitle }) => {
+  return pageTitle ? (
+    <>
+      <PageHeading title={pageTitle} />
+      {renderProbes()}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:27]: this method does not depend on any conditions or variables. It would be a better idea to just call the three components here itself.

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx [hunk 24]
```diff
@@ -0,0 +1,110 @@
...
+          style={{ maxWidth: '100%' }}
+          helpText={
+            'How many consecutive successes for the probe to be considered successful after having failed.'
+          }
+        />
+      </FormSection>
+      <ActionGroupWithIcons
+        onSubmit={onSubmit}
+        onClose={onClose}
+        isDisabled={!_.isEmpty(_.get(errors, `healthChecks.${probe}`))}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:104]: use optional chaining instead of _.get

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx [hunk 25]
```diff
@@ -0,0 +1,110 @@
...
+  onSubmit: () => void;
+  onClose: () => void;
+  probe: string;
+}
+
+enum probeTypeOptions {
+  'httpGet' = 'HTTP GET',
+  'containerCommand' = 'Container Command',
+  'tcpSocket' = 'TCP Socket',
+}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:35]: no need to use string for keys
```suggestion
enum ProbeTypeOptions {
  httpGet = 'HTTP GET',
  containerCommand = 'Container Command',
  tcpSocket = 'TCP Socket',
}
```

File: frontend/packages/dev-console/src/components/health-checks/ProbeTypeForms.tsx [hunk 26]
```diff
@@ -0,0 +1,115 @@
...
+  InputField,
+  CheckboxField,
+  DropdownField,
+  getFieldId,
+  TextColumnField,
+} from '@console/shared';
+import { NameValueEditor } from '@console/internal/components/utils/name-value-editor';
+
+interface ProbeTypeFormProps {
+  ports?: object;
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/health-checks/ProbeTypeForms.tsx:15]: add a proper type for ports

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx (hunk 23 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:27]: `renderProbes` has been used at line 30 as well

File: frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+const TextColumnField: React.FC<TextColumnFieldProps> = (props) => {
+  const { required, isValid, name, label, addLabel, helpText, placeholder, isReadOnly } = props;
+  const [field] = useField<string[]>(name);
+  useFormikValidationFix(field.value);
+  const rowValues = field.value;
+  const fieldId = getFieldId(name, 'single-column');
+  const renderField = (arrayHelpers) =>
+    rowValues.map((v, idx) => {
+      return (
+        <Flex key={`${idx.toString()}`} style={{ marginBottom: 'var(--pf-global--spacer--sm)' }}>
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx:30]: In the `MultiColumnField` component, index has been sent as a key to the Row component. Also, in the formik's fieldarray example they have used index as key

**Comment by rohitkrai03** [comment]
@debsmita1 You have linting errors.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx [hunk 28]
```diff
@@ -0,0 +1,34 @@
...
+import { HealthChecksProbeType } from './health-checks-types';
+
+interface HealthChecksProps {
+  pageTitle?: string;
+}
+
+const renderProbes = () => {
+  return (
+    <>
+      <HealthCheckProbe probe={HealthChecksProbeType.ReadinessProbe} />
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:14]:
```suggestion
      <HealthCheckProbe probeType={HealthChecksProbeType.ReadinessProbe} />
```

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx (hunk 23 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:27]: Why do you need a separate way of defining page title anyways? I think it should always be a `HealthChecksSection` component which is a form section. The form using this section should be responsible of rendering it properly with page title and what not.

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.scss [hunk 29]
```diff
@@ -0,0 +1,4 @@
+.odc-heath-check__probe-form {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/ProbeForm.scss:1]: Following BEM naming conventions a form is considered as block and not an element. http://getbem.com/naming/
```suggestion
.odc-heath-check-probe-form {
```

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx [hunk 30]
```diff
@@ -0,0 +1,108 @@
...
+const getProbeTypeForm = (value: string, probe: string, ports?: { [port: number]: number }) => {
+  switch (value) {
+    case ProbeType.HTTPGET:
+      return <HTTPProbeTypeForm ports={ports} probe={probe} />;
+    case ProbeType.ContainerCommand:
+      return <CommandProbeTypeForm probe={probe} />;
+    case ProbeType.TCPSocket:
+      return <TCPProbeTypeForm ports={ports} probe={probe} />;
+    default:
+      return undefined;
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:20]:
```suggestion
      return null;
```

File: frontend/packages/dev-console/src/components/health-checks/health-check-probe-utils.ts [hunk 31]
```diff
@@ -0,0 +1,100 @@
+import { HealthChecksProbeType, ProbeType } from './health-checks-types';
+
+export const getHealthChecksProbe = (probe: string) => {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/health-check-probe-utils.ts:3]:
```suggestion
export const getHealthChecksProbeConfig = (probe: string) => {
```

File: frontend/packages/dev-console/src/components/health-checks/health-check-probe-utils.ts [hunk 32]
```diff
@@ -0,0 +1,100 @@
...
+          'A Startup probe checks if the application within the container is started. If the startup probe fails the container is killed.',
+      };
+    }
+    default:
+      return undefined;
+  }
+};
+
+export const healthChecksProbeInitialData = {
+  healthChecks: {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/health-check-probe-utils.ts:32]: Again, why do you have an object with three probe types as property but having same data as values for all of these properties?

File: frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts [hunk 33]
```diff
@@ -0,0 +1,39 @@
...
+    port: number;
+  };
+  command?: string[];
+  initialDelaySeconds: number;
+  periodSeconds: number;
+  timeoutSeconds: number;
+  successThreshold: number;
+}
+
+export interface HealthCheckProbeType {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts:35]: You have an enum with almost same name `HealthChecksProbeType`. Its very confusing.
```suggestion
export interface HealthCheckProbe {
```

File: frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts [hunk 34]
```diff
@@ -0,0 +1,39 @@
...
+  StartupProbe = 'startupProbe',
+}
+
+export enum ProbeType {
+  HTTPGET = 'httpGet',
+  ContainerCommand = 'command',
+  TCPSocket = 'tcpSocket',
+}
+
+export interface HealthCheckProbeDataType {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts:15]:
```suggestion
export interface HealthCheckProbeData {
```

File: frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts [hunk 35]
```diff
@@ -0,0 +1,39 @@
+import { NameValuePair } from '@console/shared';
+
+export enum HealthChecksProbeType {
+  ReadinessProbe = 'readinessProbe',
+  LivenessProbe = 'livenessProbe',
+  StartupProbe = 'startupProbe',
+}
+
+export enum ProbeType {
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts:9]: What is the difference between `ProbeType` and `HealthChecksProbeType` both seems to be probe types.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx (hunk 23 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:27]: @rohitkrai03 Yes, we are always going to use this for health checks form section. So, here are two scenarios. For add flow, we need form section title and for the health check page, we don't need form section title as we already have page heading.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx [hunk 36]
```diff
@@ -0,0 +1,34 @@
...
+
+const HealthChecks: React.FC<HealthChecksProps> = ({ pageTitle }) => {
+  return pageTitle ? (
+    <>
+      <PageHeading title={pageTitle} />
+      {renderProbes()}
+    </>
+  ) : (
+    <FormSection title="Health Checks">{renderProbes()}</FormSection>
+  );
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:31]:
```suggestion
     return showTitle ? (
    <FormSection title="Health Checks">{renderProbes()}</FormSection>
  ) : (
    <>{renderProbes()}</>
  );
```
Change the ```pageTitle``` prop to ```showTitle``` and make it a boolean type.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx (hunk 23 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:27]: Right, `FormSection` has `title` as an optional prop. Just make use of that. If a title is sent from the parent it will render it if not then no title will be rendered.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx (hunk 36 above)
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/HealthChecks.tsx:31]: Based on my comments above you can simply do this -
```suggestion
const HealthChecksSection: React.FC<HealthChecksSectionProps> = ({ title }) => (
  <FormSection title={title}>
    <HealthCheckProbe probe={HealthChecksProbeType.ReadinessProbe} />
    <HealthCheckProbe probe={HealthChecksProbeType.LivenessProbe} />
    <HealthCheckProbe probe={HealthChecksProbeType.StartupProbe} />
  </FormSection>
);
```

File: frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts (hunk 35 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/health-checks-types.ts]: changed `ProbeType` to `RequestType`

File: frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx [hunk 37]
```diff
@@ -0,0 +1,120 @@
...
+          label="Failure Threshold"
+          style={{ maxWidth: '100%' }}
+          helpText={'How many times the probe will try starting or restarting before giving up.'}
+        />
+        <DropdownField
+          name={`healthChecks.${probeType}.data.requestType`}
+          label="Type"
+          items={RequestTypeOptions}
+          title={RequestType.HTTPGET}
+          onChange={onRequestTypeChange}
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx:69]: No need to send onChange handler explicitly in this case. DropdownField handles that.

File: frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx [hunk 38]
```diff
@@ -0,0 +1,92 @@
...
+import { MinusCircleIcon } from '@patternfly/react-icons';
+import { global_disabled_color_200 as disabledColor } from '@patternfly/react-tokens';
+import { InputField, useFormikValidationFix } from '@console/shared';
+import MultiColumnFieldFooter from './multi-column-field/MultiColumnFieldFooter';
+import { getFieldId } from './field-utils';
+import { FieldProps } from './field-types';
+
+export interface TextColumnFieldProps extends FieldProps {
+  required?: boolean;
+  isValid?: boolean;
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/components/formik-fields/TextColumnField.tsx:13]: Why are you taking `isValid` prop from parent? You should make use of errors and touched object from `useField` like all other fields.

File: frontend/packages/console-shared/src/components/formik-fields/BaseInputField.tsx [hunk 39]
```diff
@@ -0,0 +1,50 @@
...
+        ...props,
+        value: field.value || '',
+        id: fieldId,
+        label,
+        isValid,
+        'aria-describedby': `${fieldId}-helper`,
+        onChange: (value, event) => {
+          field.onChange(event);
+          onChange && onChange(event);
+        },
```
Comment (sahil143) [inline frontend/packages/console-shared/src/components/formik-fields/BaseInputField.tsx:44]:
```suggestion
      {children({
        ...field,
        ...props,
        value: field.value || '',
        id: fieldId,
        label,
        isValid,
        validated=!isValid ? ValidatedOptions.error : validated,
        'aria-describedby': `${fieldId}-helper`,
        onChange: (value, event) => {
          field.onChange(event);
          onChange && onChange(event);
        },
```
@debsmita1 you missed the validated prop on TextInput i.e children. which led to change in different css style and failure of integration test.
Integration test check for pf-m-success class on the text input after the validation of deploy image. and this class is added only when validated on TextInput is success(ValidatedOptions.success).

Comment (debsmita1) [reply frontend/packages/console-shared/src/components/formik-fields/BaseInputField.tsx]: Thank you for pointing this out
